First commit

achiu17 opened the pull request:
Haven't implemented much yet...still figuring out what to do exactly. So far I've only implemented a function to parse the file for values that we want (APFD, time, etc.).

winglam commented on dt-impact-tracer/src/edu/washington/cs/dt/impact/Main/ResultsParser.java:
```diff
@@ -3,15 +3,150 @@
 import java.util.Arrays;
 import java.util.List;
+import java.util.Scanner;
+import java.io.*;
```
Instead of importing the entire package, you should be able to import only the classes you need.
Ctrl + Shift + O  in Eclipse should automatically fix this for you.

winglam commented on dt-impact-tracer/src/edu/washington/cs/dt/impact/Main/ResultsParser.java:
```diff
@@ -3,15 +3,150 @@
+    public static void main(String[] args) { //args: technique, directory...optional args
         List<String> argsList = new ArrayList<String>(Arrays.asList(args));
-        System.out.println(argsList);
+        //System.out.println(argsList);
```
You can just remove this line.

winglam commented on dt-impact-tracer/src/edu/washington/cs/dt/impact/Main/ResultsParser.java:
```diff
@@ -3,15 +3,150 @@
 public class ResultsParser {
 
+
+    public static final String PARALLELIZATION = "New order time:"; //name of line to get values from in Parallelization technique
```
Instead of putting the comments for this variable after the declaration, please put it above it.
The reason for this is to somewhat enforce that each line should be no more than 100 characters.
This applies to the APFD_VALUE variable too.

winglam commented on dt-impact-tracer/src/edu/washington/cs/dt/impact/Main/ResultsParser.java:
```diff
@@ -3,15 +3,150 @@
+     * A method to search a file for a keyword and return the value that follows that keyword
+     * @return the data value without any leading or trailing whitespaces
+     */
+    public static String parseFile(File file, String keyword) throws FileNotFoundException{
```
This can probably be changed to a private method.

winglam commented on dt-impact-tracer/src/edu/washington/cs/dt/impact/Main/ResultsParser.java:
```diff
@@ -3,15 +3,150 @@
-        System.out.println(argsList);
+        //System.out.println(argsList);
+
+        String technique = args[0];
```
Instead of assuming that the first argument is always technique and the second one is always the directory, look for where in the argument list is a argument like "-techniqueName" and then parse the string after that argument.
For more examples of this see the Wrapper class.
https://github.com/winglam/dependent-tests-impact/blob/master/dt-impact-tracer/src/edu/washington/cs/dt/impact/Main/Wrapper.java

winglam commented on dt-impact-tracer/src/edu/washington/cs/dt/impact/Main/ResultsParser.java:
```diff
@@ -3,15 +3,150 @@
+        File directory = new File(directoryName);
+        File[] fList = directory.listFiles();
+        
+        for(File file: fList){
```
It seems you have not applied the saved actions and formatting guidelines I sent you for Eclipse. 
Please do so soon.

achiu17 commented:
Second commit...I've implemented a function to parse through the flags to get and return the corresponding arguments. The next step is figuring out the best way to generate the LaTeX.

winglam commented on dt-impact-tracer/src/edu/washington/cs/dt/impact/Main/ResultsParser.java:
```diff
@@ -1,17 +1,169 @@
+
+        String projectName = getArgName(argsList, "-project");
+
+        String testType = getArgName(argsList, "-testType");
```
If you directly get the String this way and start using it then the user will not be able to specify more than one configuration and as a result this class will not be able to directly generate figures 17, 18, and 19. You can either check for a delimiter when you get the configurations or you can directly just get the configurations from the files.

From doing the former option to generate figure 17 you would have to specify something like "-projects crystal,jfreechart,jodatime,synoptic,xml_security". If you directly parse the files from the directory, whatever projects it sees, it will generate a row/column for in the table.

achiu17 commented:
I've tested my code with a few files and it works with the exception of a few bugs. Instead of having 2 "\" at the end in the output file, there's only one "\". Additionally, I am going to rewrite the last chunk of code that has to do with writing to the output files by putting it into a function. Everything else seems to be working fine when I tested it with a few files. Will continue to debug and fix issues as I find them

achiu17 commented:
I've fixed the bugs since my last commit and cleaned up my code in the last part by putting it into a function. After some testing, I believe my program is able to generate the needed LaTeX for figures 17 and 18 (with the exception of the percentages). One thing that will probably need to be changed is where the LaTeX is getting written to.

winglam commented on dt-impact-tracer/src/edu/washington/cs/dt/impact/Main/ResultsParser.java:
```diff
@@ -1,17 +1,527 @@
-package edu.washington.cs.dt.impact.Main;
 
+// package edu.washington.cs.dt.impact.Main;
```
Why is this commented out? Does this compile?

winglam commented on dt-impact-tracer/src/edu/washington/cs/dt/impact/Main/ResultsParser.java:
```diff
@@ -1,17 +1,527 @@
 import java.util.List;
+import java.util.Scanner;
+
+class Project {
```
This should be in it's own class. You can make a new class for this under src/edu/washington/cs/dt/impact/data

achiu17 commented:
Implemented Figure 19 generator...still need to do the percentages for Figure 18.

winglam commented on dt-impact-tracer/src/edu/washington/cs/dt/impact/Main/ResultsParser.java:
```diff
@@ -1,17 +1,496 @@
+            }
+        }
+
+        // generate LaTeX for the orig-written and automatic test suites
```
"orig-written" should be "human-written"

winglam commented on dt-impact-tracer/src/edu/washington/cs/dt/impact/Main/ResultsParser.java:
```diff
@@ -1,17 +1,496 @@
+
+        String origOutputFilename = outputDirectoryName + "/";
+        String autoOutputFilename = outputDirectoryName + "/";
+        if (proj_orig_arrayList.get(0).isFig17()) {
```
It seems that if a project is true for isFig17 but is also true for isFig18 and isFig19 then only figure 17 results will be outputted.
This should be changed such that _all_ figures that is true for a project will be outputted.

achiu17 replied:
I thought projects would only be of the same type (like Figure 17, 18, or 19) in one directory. If not, I can adjust it

winglam replied:
It's fine to make this assumption but it should be clarified in the comments that this is the case. 
However it seems that the code supports outputting multiple figures if they were all in one directory, if this is the case then you should adjust it.

achiu17 commented:
Finished implementing ResultsParser to generate Figures 17, 18, and 19. Tested for a couple files and seems to work...will continue to test and refine code.
